source/fundamentals/connection/tls.txt

```diff
@@ -0,0 +1,191 @@
+                .applyToSslSettings(builder -> 
+                    builder.enabled(true))
+                .build();
+         MongoClient client = MongoClients.create(settings);
+
+.. _tls_configure-certificates:
+
+Configure Certificates
+----------------------
+
+Java applications that serve TLS/SSL requests require access to
+cryptographic certificates that prove identity for the application
+itself as well as other applications with which the application
+interacts. To configure access to these certificates in your application,
+you should use the JVM Trust Store and the JVM Key Store.
+
+.. _tls-configure-jvm-truststore:
+
+Configure the JVM Trust Store
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The JVM trust store saves certificates that securely identify other
+applications with which your Java application interacts. Using these
+certificates, your application can prove that the connection to another
+application is genuine and secure from tampering by third parties.
+
+An application that serves TLS/SSL requests needs to configure two
+JVM system properties to ensure that the client is able to validate
+the TLS/SSL certificate presented by a connected MongoDB instance.
+
+- ``javax.net.ssl.trustStore``: the path to a trust store containing the
+  certificate of the signing authority
+
+- ``javax.net.ssl.trustStorePassword``: the password to access the trust
+  store defined in ``javax.net.ssl.trustStore``
+
+You can create a trust store with the `keytool
+<http://docs.oracle.com/javase/8/docs/technotes/tools/unix/keytool.html>`__
+command line tool provided as part of the JDK:
+
+.. code-block:: console
+
+   keytool -importcert -trustcacerts -file <path to certificate authority file>
+            -keystore <path to trust store> -storepass <password>
+
+Configure the JVM Key Store
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The JVM key store saves certificates that securely identify your Java
+application to other applications. Using these certificates, other
+applications can prove that the connection to your application is
+genuine and secure from tampering by third parties.
+
+An application that serves TLS/SSL requests needs to set two JVM system
+properties to ensure that the client presents a TLS/SSL certificate to
+the MongoDB server:
+
+- ``javax.net.ssl.keyStore``: the path to a key store containing the client's
+  TLS/SSL certificates
+
+- ``javax.net.ssl.keyStorePassword``: the password to access the key store
+  defined in ``javax.net.ssl.keyStore``
+
+You can create a key store with the `keytool
+<http://docs.oracle.com/javase/8/docs/technotes/tools/unix/keytool.html>`__
+or `openssl <https://www.openssl.org/docs/apps/openssl.html>`__ command
+line tools.
+
+For more information on configuring a Java application to use TLS/SSL,
+please refer to the `JSSE Reference Guide
+<http://docs.oracle.com/javase/8/docs/technotes/guides/security/jsse/JSSERefGuide.html>`__.
+
+.. _tls-disable-hostname-verification:
+
```

Comment:
Can we add a debugging tls/ssl section or note block with a link to : https://docs.oracle.com/javase/8/docs/technotes/guides/security/jsse/ReadDebug.html to help users debugging any issues.  Tls on java is a real pain so mentioning setting the system property: `-Djavax.net.debug=all` to generate debugging output will really help users out.

Reply:
Great suggestion, added a note at the top of the TLS page!